src/target/riscv/riscv.c

```diff
@@ -4001,7 +4012,11 @@ static int riscv_openocd_step_impl(struct target *target, int current,
 		LOG_TARGET_ERROR(target, "Unable to step rtos hart.");
 	}
 
-	register_cache_invalidate(target->reg_cache);
+	if (riscv_reg_any_dirty(target, LOG_LVL_ERROR))
+		LOG_TARGET_ERROR(target,
+				"BUG: registers should have been flushed by this point.");
```

Comment:
If you don't mind, I would make it very explicit to the code reader what this error means:

```suggestion
		// If this happens, it means there is a bug in the previous register-flushing algorithm:
		// not all registers were flushed back to the target prior to single-step.
		LOG_TARGET_ERROR(target,
				"BUG: registers should have been flushed by this point.");
```

Reply:
Addressed.